### PR TITLE
ADR for removing Aggregator layer

### PR DESCRIPTION
**Pull request by pritipath**
Issue Number: (Link to Github Issue or Azure Dev Ops Task/Story)

## Summary
With the addition of the GSF library, aggregator layer is an extra overhead in the architecture. The proposal is to remove the Aggregator layer and migrate the data manipulation logic from aggregators to the handlers present in the GSF library.

## Changes

ADR for removing Aggregator layer

## Checklist

- [ ] Local Tests Passing?
- [ ] CICD and Pipeline Tests Passing?
- [ ] Added any new Tests?
- [ ] Documentation Updates Made?
- [ ] Are there any API Changes? If yes, please describe below.
- [ ] This is not a breaking change. If it is, please describe it below.

## Are there API Changes?
If yes, what are the expected API Changes? Please link to an API-Comparison workflow with the API Diff.

## Is this a breaking change?
If yes, what workflow does this break? 

## Anything else?
Other comments, collaborators, etc.
> Please follow [GitHub's suggested syntax](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) to link Pull Requests to Issues via keywords

This PR Closes #<issue_number>

### CODE REVIEW

File: docs/architecture/decisions/0013-remove-aggregator-layer.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# 0013. Remove Aggregator layer from the Carbon Aware Architechture
+
+## Status
+
+Proposed
+
+## Context
+The Carbon Aware SDK provides a public library to retrieve carbon emissions data. The library consists of public handler interfaces and data models that can be consumed by any consumer layer like WebAPI, CLI, Azure Functions etc.
+With the inclusion of these handlers, the Aggregator layer has now become superfluos. It is an extra pass through layer that contains business logic for data manipulation.
```
Comment (juzuluag) [inline docs/architecture/decisions/0013-remove-aggregator-layer.md:9]:
```suggestion
With the inclusion of these handlers, the Aggregator layer has now become obsolete. It is has been replaced by the handlers, which are more aligned with the carbon intensity feature.
```

File: docs/architecture/decisions/0013-remove-aggregator-layer.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+## Sequence diagram
+
+![WebApi Screenshot](../../images/end-end-tiers.png)
+
+
+## Benefits
+
+- Clear access boundaries defined
+- High maintainability since there is less code to maintain
```
Comment (juzuluag) [inline docs/architecture/decisions/0013-remove-aggregator-layer.md:26]:
```suggestion
- Less code to maintain.
```

File: docs/architecture/decisions/0013-remove-aggregator-layer.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+- Smaller size of packaged binary 
+
```
Comment (juzuluag) [inline docs/architecture/decisions/0013-remove-aggregator-layer.md:28]:
```suggestion
- Clear separation of concerns between different layers
```

**Review by juzuluag — CHANGES_REQUESTED** [review]
Overall looks good. Just a few suggested changes 👍

**Review by JenMadiedo — APPROVED** [review]
Looks good to me. Would it be worth adding some labels in the swimlane to further specify what the layers look like now (Consumers tier are WebAPI, CLI, Library through nuget package etc..), Business layer is the library etc.

**Review by juzuluag — APPROVED** [review]
👍

**Review by Kpopovs1 — APPROVED** [review]

**Review by gfmatthews — APPROVED** [review]
Great job on this!

**Review by vaughanknight — APPROVED** [review]
This looks good, and I assume we will be looking to approve.  Will confirm after today's GSF meeting.

**Pull request merged**